diff --git a/core/autoload/config.el b/core/autoload/config.el
index 1e643cf49..501cbfba4 100644
--- a/core/autoload/config.el
+++ b/core/autoload/config.el
@@ -23,7 +23,7 @@
 (defun doom/find-file-in-private-config ()
   "Search for a file in `doom-private-dir' inside nixos-config."
   (interactive)
-  (doom-project-find-file doom-private-dir))
+  (doom-project-find-file "@private@"))
 
 ;;;###autoload
 (defun doom/goto-private-init-file ()
diff --git a/modules/app/rss/config.el b/modules/app/rss/config.el
index c5657eec8..34f303420 100644
--- a/modules/app/rss/config.el
+++ b/modules/app/rss/config.el
@@ -18,8 +18,8 @@ easier to scroll through.")
 (use-package! elfeed
   :commands elfeed
   :init
-  (setq elfeed-db-directory (concat doom-local-dir "elfeed/db/")
-        elfeed-enclosure-default-dir (concat doom-local-dir "elfeed/enclosures/"))
+  (setq elfeed-db-directory (concat doom-cache-dir "elfeed/db/")
+        elfeed-enclosure-default-dir (concat doom-cache-dir "elfeed/enclosures/"))
   :config
   (setq elfeed-search-filter "@2-week-ago "
         elfeed-show-entry-switch #'pop-to-buffer
