diff --git a/core/autoload/config.el b/core/autoload/config.el
index 1e643cf49..501cbfba4 100644
--- a/core/autoload/config.el
+++ b/core/autoload/config.el
@@ -23,7 +23,7 @@
 (defun doom/find-file-in-private-config ()
   "Search for a file in `doom-private-dir' inside nixos-config."
   (interactive)
-  (doom-project-find-file doom-private-dir))
+  (doom-project-find-file "@private@"))
 
 ;;;###autoload
 (defun doom/goto-private-init-file ()
