diff --git a/core/core.el b/core/core.el
index c8cfb0495..d8ed107bd 100644
--- a/core/core.el
+++ b/core/core.el
@@ -92,20 +92,20 @@ envvar will enable this at startup.")
 (defconst doom-local-dir
   (if-let (localdir (getenv "DOOMLOCALDIR"))
       (expand-file-name (file-name-as-directory localdir))
-    (concat doom-emacs-dir ".local/"))
+      "@local@/")
   "Root directory for local storage.
 
 Use this as a storage location for this system's installation of Doom Emacs.
 These files should not be shared across systems. By default, it is used by
 `doom-etc-dir' and `doom-cache-dir'. Must end with a slash.")
 
-(defconst doom-etc-dir (concat doom-local-dir "etc/")
+(defconst doom-etc-dir "~/.local/doom/"
   "Directory for non-volatile local storage.
 
 Use this for files that don't change much, like server binaries, external
 dependencies or long-term shared data. Must end with a slash.")
 
-(defconst doom-cache-dir (concat doom-local-dir "cache/")
+(defconst doom-cache-dir "~/.cache/doom/"
   "Directory for volatile local storage.
 
 Use this for files that change often, like cache files. Must end with a slash.")
